### PR TITLE
Fix json parsing behavior

### PR DESCRIPTION
**Pull request by piiswrong**
@tqchen

### CODE REVIEW

**Comment by jermainewang** [comment]
BTW, why SaveJSON returns an empty graph rather than the original graph?

File: src/core/symbolic.cc [hunk 1]
```diff
@@ -10,7 +10,7 @@
 namespace nnvm {
 
 namespace symbol_constants {
-const char *kNamespaceSeparator = "_";
+const char *kNamespaceSeparator = "$";
```
Comment (tqchen) [inline src/core/symbolic.cc]: will this make things not backward compatible?

Comment (piiswrong) [inline src/core/symbolic.cc]: saved json will be upgraded
the only thing that's broken is list_attr(recursive=True)
I raised a warning when it's called and replaced it with attr_dict

Comment (piiswrong) [inline src/core/symbolic.cc]: If we really want compatibility we can just replace $ with _ in frontend (the reverse is not possible)
but I think using _ as separation is really a bad idea and should be banned

Comment (tqchen) [inline src/core/symbolic.cc]: As long as the front-end behavior is fixed, I am fine with this change. Consider use "/" instead for separator, which is more typical

Comment (piiswrong) [inline src/core/symbolic.cc]: I wanted to use something no one would use so that its safe to split on it

Comment (piiswrong) [inline src/core/symbolic.cc]: also this is not exposed in frontend

**Pull request merged**